src/avalanchemq/client/client.cr

```diff
@@ -191,6 +191,10 @@ module AvalancheMQ
 
     def send(frame : AMQP::Frame) : Bool
       return false if closed?
+      unless frame.channel.zero? || @channels[frame.channel]?.try &.running?
```

Comment:
Inefficient, in the cases where we deliver from a `Channel`, then we can check `@running` already there, this implementation always requires an extra hash lookup.